sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Util.scala

```diff
@@ -346,4 +346,17 @@ private[sql] object CatalogV2Util {
       }
     }
   }
+
+  def failNullType(dt: DataType): Unit = {
+    if (NullType.containsNullType(dt)) {
+      throw new AnalysisException(
+        "Cannot create tables with VOID type.")
+    }
+  }
+
+  def assertNoNullTypeInSchema(schema: StructType): Unit = {
+    schema.foreach { f =>
+      failNullType(CatalystSqlParser.parseDataType(schema.catalogString))
```

Comment:
shouldn't this be `failNullType(f.dataType)`?

Reply:
Ah, yes. Remove `CatalystSqlParser.parseDataType(schema.catalogString)` first could also remove `case ("null", Nil) => NullType` then.